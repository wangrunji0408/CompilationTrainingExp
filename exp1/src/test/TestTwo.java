package test;

public class TestTwo {
    public static int frobble(int x) {
	return 8-x;
    }

    public static void main (String [] args) {
	int a=0, b=0, c=0, d;

	for (a=0; a<7; a++) {

	    for (d=0; d<7; d++) {
	        c=frobble(c);
	    }
				
	    if (a-b == 2) {
		c++;
	    } else if (c == 4) {
		break;
	    }
	}

	c++;

    }
}
