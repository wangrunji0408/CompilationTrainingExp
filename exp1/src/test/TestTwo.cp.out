Now analyzing test.TestTwo
Method: frobble
Initialization completed.
entry: {R0=NAC, T1=undef}
0 in:  {R0=undef, T1=undef}
0 out: {R0=undef, T1=undef}
1 in:  {R0=NAC, T1=undef}
1 out: {R0=NAC, T1=NAC}
2 in:  {R0=NAC, T1=NAC}
2 out: {R0=NAC, T1=NAC}
exit: {R0=NAC, T1=NAC}
Method: main
Initialization completed.
entry: {R0=NAC, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
0 in:  {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
0 out: {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
1 in:  {R0=NAC, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
1 out: {R0=NAC, R1=0, R2=undef, R3=undef, R4=undef, T5=undef}
2 in:  {R0=NAC, R1=0, R2=undef, R3=undef, R4=undef, T5=undef}
2 out: {R0=NAC, R1=0, R2=0, R3=undef, R4=undef, T5=undef}
3 in:  {R0=NAC, R1=0, R2=0, R3=undef, R4=undef, T5=undef}
3 out: {R0=NAC, R1=0, R2=0, R3=0, R4=undef, T5=undef}
4 in:  {R0=NAC, R1=0, R2=0, R3=0, R4=undef, T5=undef}
4 out: {R0=NAC, R1=0, R2=0, R3=0, R4=undef, T5=undef}
5 in:  {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
5 out: {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
6 in:  {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
6 out: {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
7 in:  {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
7 out: {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
8 in:  {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
8 out: {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
9 in:  {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
9 out: {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
10 in:  {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
10 out: {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
11 in:  {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
11 out: {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
12 in:  {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
12 out: {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
13 in:  {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
13 out: {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
14 in:  {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
14 out: {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
15 in:  {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
15 out: {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
16 in:  {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
16 out: {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
17 in:  {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
17 out: {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
18 in:  {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
18 out: {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
19 in:  {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
19 out: {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
20 in:  {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
20 out: {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
21 in:  {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
21 out: {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
22 in:  {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
22 out: {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
23 in:  {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
23 out: {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
24 in:  {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
24 out: {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
25 in:  {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
25 out: {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=0, T5=NAC}
26 in:  {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
26 out: {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
27 in:  {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
27 out: {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
28 in:  {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
28 out: {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
29 in:  {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
29 out: {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
30 in:  {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
30 out: {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
31 in:  {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
31 out: {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
32 in:  {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
32 out: {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
33 in:  {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
33 out: {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
34 in:  {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
34 out: {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
35 in:  {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
35 out: {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
36 in:  {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
36 out: {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
37 in:  {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
37 out: {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
38 in:  {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
38 out: {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
39 in:  {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
39 out: {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
40 in:  {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
40 out: {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
41 in:  {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
41 out: {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
42 in:  {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
42 out: {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
43 in:  {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
43 out: {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
44 in:  {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
44 out: {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
45 in:  {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
45 out: {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
46 in:  {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
46 out: {R0=undef, R1=undef, R2=undef, R3=undef, R4=undef, T5=undef}
47 in:  {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
47 out: {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
48 in:  {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
48 out: {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
49 in:  {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
49 out: {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
50 in:  {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
50 out: {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
51 in:  {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
51 out: {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
52 in:  {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
52 out: {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
exit: {R0=NAC, R1=NAC, R2=0, R3=NAC, R4=NAC, T5=NAC}
Method: <init>
Initialization completed.
entry: {R0=NAC, T-1=undef}
0 in:  {R0=undef, T-1=undef}
0 out: {R0=undef, T-1=undef}
1 in:  {R0=NAC, T-1=undef}
1 out: {R0=NAC, T-1=undef}
2 in:  {R0=NAC, T-1=undef}
2 out: {R0=NAC, T-1=undef}
3 in:  {R0=NAC, T-1=undef}
3 out: {R0=NAC, T-1=undef}
exit: {R0=NAC, T-1=undef}
